specs/core/0_beacon-chain.md

```diff
@@ -639,6 +650,19 @@ def slot_to_epoch(slot: SlotNumber) -> EpochNumber:
     return slot // EPOCH_LENGTH
 ```
 
+### `get_previous_epoch`
+
+```python
+def get_previous_epoch(state: BeaconState) -> EpochNumber:
+    """`
+    Return the previous epoch of the given ``state``.
+    If the current epoch is  ``GENESIS_EPOCH``, return ``GENESIS_EPOCH``.
```

Comment:
This logic doesn't match the implementation.

Reply:
This is diff than our implementation, I wrote the code before putting this helper function in spec. Let's update our implementation, will open an issue

Reply:
What i am saying is this comment says 
```
 If the current epoch is  ``GENESIS_EPOCH``, return ``GENESIS_EPOCH``.
```
but the pseudo code below says
```
 if the current epoch X is less than GENESIS_EPOCH, return X
```

Reply:
The code actually implemented the same functionality as the comment but the semantics were way different. Updated it for clarity.